### ISSUE TITLE
Using the sidebar overwrites files. 

### ISSUE THREAD

**Issue by eliocamp** [lainsce/quilter#254]
Steps to replicate the issue:

1. Open Quilter (with no documents in sidebar - it replicates either way, but this is simpler to show). 
2. Create a new document. (Doc1)
3. Write something: "Test text".
4. Create another new document. (Doc2)

Issue1: Doc2 is not blank, but it start with "Test text". 

5. Replace "Test text" in the Doc2 with anything else: "Important text". 
6. Select Doc1. 

Issue2: Doc1 now "Important text" instead of "Test text". 

7. Change text in Doc1 to another thing: "Let's test again"
8. Select Doc2. It shows (correctly), "Important text". 
9. Select Doc1 again.

Yet again, it's contents are replaced by "Important text". 

This is bad as it is, but the autosave feature means that it also overwrites actual files, automatically making you lose potentially important information or progress. 

Here's an animation. Note that because I had already done the steps detailed before, the first new document already has content!

![Peek 2019-05-30 11-28](https://user-images.githubusercontent.com/8617595/58640017-881d3480-82ce-11e9-9d16-218129dd5e7f.gif)

I'm using Quilter 1.8.5 on Juno fully updated.

**Comment by lainsce** [comment]
When Quilter has any "New Document" tab open, it's using a cache file and not any actual file on disk, so if you have more than one like that, it's actually all connected to the cache. If you open actual files, any saved test files and do the same operations as you describe, does it do the same thing?

**Comment by eliocamp** [comment]
So then having more than one unsaved file is not supported?

Using actual files, I have similar issues:

![Peek 2019-05-30 14-25](https://user-images.githubusercontent.com/8617595/58651246-23220880-82e7-11e9-89eb-7b28ac8ab5b2.gif)

**Comment by lainsce** [comment]
It's a relic of when Quilter wasn't multifile, I'm working towards fixing this.

**Comment by lainsce** [comment]
Should be fixed now on 1.9.0

**Comment by lainsce** [comment]
> It seems to me that quilters don't save every change immediately, but only after a certain period of time. Is that true?

It is true, but changing between documents should also trigger a save.

> Where are the close buttons next to the individual files?

To solve the bug of corrupted files, I decided it would be best to treat all open files as a "session". Once you're done with the session, you clean it with the button and start a new "session".

**Comment by 4jNsY6fCVqZv** [comment]
> It is true, but changing between documents should also trigger a save.

Well, that'd be great if it did. Unfortunately, it doesn't look like it's doing that right now.  Can you please fix that?

**Comment by lainsce** [comment]
I will as soon as I can!

**Comment by 4jNsY6fCVqZv** [comment]
> I will as soon as I can!

Thank you. And also thanks for your great work on Quilter!

> Open an issue so that I'm reminded of that.

You think it makes sense to reopen this issue here?
Otherwise, what should I write so that you remember well?

**Comment by 4jNsY6fCVqZv** [comment]
btw. I've noticed some other issues. Does that suit you if they all end up on the issue tracker?

**Comment by lainsce** [comment]
All issues you find would be useful to be open as individual issues

**Comment by 4jNsY6fCVqZv** [comment]
> Also, not necromancing this issue for something that's not related to the original issue.

Well, it's identical to what you see in this comment https://github.com/lainsce/quilter/issues/254#issuecomment-497413280
That's why I thought it could be part of this issue and stay part of it.

**Comment by lainsce** [comment]
It's better to do new issues than old ones, even if that comment is here. It's better for me so I can prioritize your issues.

**Comment by 4jNsY6fCVqZv** [comment]
>  It seems to me that quilters don't save every change immediately, but only after a certain period of time. Is that true?
It is true

Is that intentional?
Otherwise I would want a direct save without a time out. From the first changed character on.
How does elementary Code do that?

**Comment by lainsce** [comment]
The thing is, Quilter does not do saves immediately because that would kill your HDD/SSD faster than ice in the asphalt. It's a way to alleviate disk usage. Code doesn't do it because reasons, I guess.

**Comment by 4jNsY6fCVqZv** [comment]
> The thing is, Quilter does not do saves immediately because that would kill your HDD/SSD faster than ice in the asphalt.

Why would it do that? For example, if I rewrite 100 characters and remove 120, the need is less, isn't it? Unless only 220 additional characters are written and old characters are not removed from memory at the same time.

> Code doesn't do it because reasons, I guess.

So far, my perception of Code is that the balance here is very well adjusted. Do you think you could check this with the Quilter code base and make adjustments if possible and make sense?

**Comment by lainsce** [comment]
> If I rewrite 100 characters and remove 120

Then you have written 220 bytes total for the whole operation, and it reads every single byte to be shown on Quilter. Imagine doing that on bigger texts. The amount would skyrocket!
